data_adapter_oemof/calculations.py

```diff
@@ -34,3 +42,283 @@ def get_name(*args, counter=None):
+        capacity_tra_connection_max
+        flow_share_max_<commodity>
+        sto_cycles_max
+        sto_max_timeseries
+
+        Returns
+        -------
+
+        """
+        max_value = 9999999999999
+        min_value = 0
+
+        min = ["capacity_p_min", "capacity_e_min", "capacity_w_min", "flow_share_min_"]
+
+        max = [
+            "potential_annual_max",
+            "capacity_p_max",
+            "capacity_e_max",
+            "capacity_w_max",
+            "availability_timeseries_max",
+            "capacity_tra_connection_max",
+            "flow_share_max_",
+            "sto_cycles_max",
+            "sto_max_timeseries",
+            "capacity_p_abs_new_max",
+            "capacity_e_abs_new_max",
+            "capacity_w_abs_new_max",
+        ]
+
+        for column in group_df.columns:
+            if column in ["method", "source", "comment", "bandwidth_type"]:
+                continue
+
+            """
+            Following is a check whether nans can be filled.
+
+            Commented check for columns that are faulty and need to be changed
+            Commented Error for incomplete columns as we dont know where it may cause errors yet
+
+            """
+            if column in max:
+                group_df[column] = group_df[column].fillna(max_value)
+            elif column in min:
+                group_df[column] = group_df[column].fillna(min_value)
+
+        return group_df
+
+    def find_and_replace_irrelevant_data(group_df: pd.DataFrame) -> pd.DataFrame:
+        """
+        Finds and replaces irrelevant Data.
+
+        Searches for where investment is allowed
+            - If no Investment is allowed and there is no period above where investment was allowed
+                -> Data is replaced by pandas ffil/bfill
+            - If no Investment is allowed and there have been periods with investment before
+                -> Data is replaced by pandas ffil/bfill
+
+        Searches for decomissioned Processes
+            - If capacity of a process is 0 other nan data is replaced by mean.
+
+        Parameters
+        ----------
+        group_df
+
+        Returns
+        -------
+
+        """
+
+        capacity_columns = [
+            "capacity_p_inst_0",
+            "capacity_e_inst_0",
+            "capacity_w_inst_0",
+            "capacity_tra_inst_0",
+        ]
+
+        invest_zero = [
+            "capacity_p_abs_new_max",
+            "capacity_e_abs_new_max",
+            "capacity_w_abs_new_max",
+            "capacity_p_max",
+            "capacity_e_max",
+            "capacity_w_max",
+        ]
+        # no investment in decommissioning processes
+        # no capacity can be set on investment objects
+        if not any([x in capacity_columns for x in group_df.columns]):
+            for invest_column in invest_zero:
+                if invest_column in group_df.columns:
+                    non_investment_indices = group_df[invest_column] == 0
```

Comment:
What does this check means? If whole column is zero? Then filling would be senselesss?!
Or does it have another meaning? difficult to understand.

Reply:
Line 300 returns all indices that where invest column is 0. I can use that indices to only fill the rows where investment is not allowed.

Basically I want to Iterate all columns where investment allowances are set and check if for any year investment is 0 and then fill all nan values on other columns where investment is 0.

Reply:
I see! 
But I think this is only working as you first fill all missing values and slice missing rows afterwards (see my suggestion below. It shouldn't work after switching order of filling and slicing, as then the filling function does not see the "good" values it needs to fill up missing values. (you can try though)
In order to fix this, I would recommend to fill complete columns and not to slice them in the first way.

Reply:
So in other words just fill all missing values regardless of investment/capacity figures since we are expecting "good" data anyways?

Reply:
no - i would still figure out which columns to fill.
But I would not check for zeros.
Did you know that fill only replaces NAN values and not zeros? -> https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.ffill.html#pandas.DataFrame.ffill

Reply:
I adapted using DataFrame.mask() instead of my previouos logic, I hope it is better readable and does the same job here.